src/components/engine/engine-instances-table.tsx

```diff
@@ -372,17 +370,19 @@ const RemoveModal = ({
   refetch: () => void;
 }) => {
   const toast = useToast();
+  const { mutate: removeFromDashboard } = useEngineRemoveFromDashboard();
+  const { mutate: removeCloudHosted } = useEngineRemoveCloudHosted();
   const { onClose } = disclosure;
 
+  const form = useForm<RemoveCloudHostedInput>({
+    defaultValues: {
+      instanceId: instance.id,
+    },
+  });
+
   const onClickRemove = async () => {
     try {
-      const res = await fetch(`${THIRDWEB_API_HOST}/v1/engine/${instance.id}`, {
-        method: "DELETE",
-        credentials: "include",
-      });
-      if (!res.ok) {
-        throw new Error(`Unexpected status ${res.status}`);
-      }
+      await removeFromDashboard(instance.id);
```

Comment:
same here, just move the success / error case to the mutate options callbacks `onSuccess` and `onError`